### PR TITLE
Add AllocationId to telemetry metadata

### PR DESCRIPTION
**Pull request by zhiyuanliang-ms**
reference: https://github.com/Azure/azure-sdk-for-python/pull/37840/files#diff-a38814ad69b5c0a6d82fe7946285bb3408209336deb2ba83d8b8808d13ea4657

### CODE REVIEW

File: src/AzureAppConfigurationImpl.ts [hunk 1]
```diff
@@ -613,6 +637,118 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
...
+                const percentileAllocation: string[] = [];
+                for (const percentile of sortedPercentileList) {
+                    variantsForExperiementation.push(percentile[VARIANT_KEY_NAME]);
+                    percentileAllocation.push(`${percentile[FROM_KEY_NAME]},${base64Helper(percentile[VARIANT_KEY_NAME])},${percentile[TO_KEY_NAME]}`);
+                }
+                rawAllocationId += percentileAllocation.join(";");
+            }
+        }
+
+        if (featureFlag[ALLOCATION_KEY_NAME] === undefined || (variantsForExperiementation.length === 0 && featureFlag[ALLOCATION_KEY_NAME][SEED_KEY_NAME] === undefined)) {
```
Comment (rossgrambo) [inline src/AzureAppConfigurationImpl.ts:676]:
```suggestion
        if (variantsForExperiementation.length === 0 && featureFlag[ALLOCATION_KEY_NAME][SEED_KEY_NAME] === undefined) {
```

We're doing this same check multiple times so I'd rather combine them. Added a suggestion earlier to do so ^

File: src/AzureAppConfigurationImpl.ts [hunk 2]
```diff
@@ -546,10 +567,12 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
 
         if (featureFlag[TELEMETRY_KEY_NAME] && featureFlag[TELEMETRY_KEY_NAME][ENABLED_KEY_NAME] === true) {
             const metadata = featureFlag[TELEMETRY_KEY_NAME][METADATA_KEY_NAME];
+            const allocationId = await this.#generateAllocationId(featureFlag);
```
Comment (rossgrambo) [inline src/AzureAppConfigurationImpl.ts:570]:
```suggestion
            const allocationId = "";
            if (featureFlag[ALLOCATION_KEY_NAME] !== undefined) {
                allocationId = await this.#generateAllocationId(featureFlag);
            }
```

If we do this check here we can remove it from other places

File: src/AzureAppConfigurationImpl.ts [hunk 3]
```diff
@@ -613,6 +637,118 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
...
         return featureFlagReference;
     }
+
+    async #generateAllocationId(featureFlag: any): Promise<string> {
+        let rawAllocationId = "";
+        // Only default variant when enabled and variants allocated by percentile involve in the experimentation
+        // The allocation id is genearted from default variant when enabled and percentile allocation
+        const variantsForExperiementation: string[] = [];
+
+        if (featureFlag[ALLOCATION_KEY_NAME]) {
```
Comment (rossgrambo) [inline src/AzureAppConfigurationImpl.ts:647]:
```suggestion
```

Removed in my other comments

**Review by rossgrambo — APPROVED** [review]

File: src/AzureAppConfigurationImpl.ts (hunk 1 above)
Comment (zhiyuanliang-ms) [reply src/AzureAppConfigurationImpl.ts:676]: If `featureFlag[ALLOCATION_KEY_NAME]` is undefined, `featureFlag[ALLOCATION_KEY_NAME][SEED_KEY_NAME]` will throw exception.

**Review by linglingye001 — APPROVED** [review]

**Review by rossgrambo — APPROVED** [review]

**Pull request merged**